client/lcd/swagger-ui/swagger.yaml

```diff
@@ -521,9 +556,629 @@ paths:
+        404:
+          description: Not Found
+        500:
+          description: Internal Server Error
+  /stake/validators:
+    get:
+      summary: Get all validator candidates
+      tags:
+      - ICS21
+      produces:
+      - application/json
+      responses:
+        200:
+          description: OK
+        404:
+          description: Not Found
+        500:
+          description: Internal Server Error
+  /stake/validators/{validatorAddr}:
+    parameters:
+    - in: path
+      name: validatorAddr
+      description: Bech32 OperatorAddress of validator
+      required: true
+      type: string
+    get:
+      summary: Query the information from a single validator
+      tags:
+      - ICS21
+      produces:
+      - application/json
+      responses:
+        200:
+          description: OK
+        404:
+          description: Not Found
+        500:
+          description: Internal Server Error
+  /stake/pool:
+    get:
+      summary: Get the current state of the staking pool
+      tags:
+      - ICS21
+      produces:
+      - application/json
+      responses:
+        200:
+          description: OK
+        500:
+          description: Internal Server Error
+  /stake/parameters:
+    get:
+      summary: Get the current staking parameter values
+      tags:
+      - ICS21
+      produces:
+      - application/json
+      responses:
+        200:
+          description: OK
+        500:
+          description: Internal Server Error
+  /slashing/signing_info/{validator}:
+    get:
+      summary: Get sign info of given validator
+      description: Get sign info of given validator
+      produces:
+      - application/json
+      tags:
+      - ICS23
+      parameters:
+      - type: string
+        description: Bech32 validator public key
+        name: validator
+        required: true
+        in: path
+      responses:
+        '200':
+          description: OK
+        500:
+          description: Internal Server Error
+  /slashing/unjail:
+    post:
+      summary: Unjail a jailed validator
+      description: Send transaction to unjail a jailed validator
+      consumes:
+      - application/json
+      produces:
+      - application/json
+      tags:
+      - ICS23
+      parameters:
+      - in: query
+        name: simulate
+        description: if true, ignore the gas field and perform a simulation of a transaction, but don't broadcast it
+        required: false
+        type: boolean
+      - in: query
+        name: generate_only
+        description: if true, build an unsigned transaction and write it back
+        required: false
+        type: boolean
+      - description: ''
+        name: UnjailBody
+        in: body
+        required: true
+        schema:
+          type: object
+          properties:
+            base_req:
+              "$ref": "#/definitions/BaseReq"
+            validator_addr:
+              "$ref": "#/definitions/ValidatorAddress"
+      responses:
+        200:
```

Comment:
if we do restful, we are missing a 404 here

Reply:
What happens in the case that the validator address is wrong and the validator can't be found?

Reply:
If the parameter in the request is invalid, for instance, invalid validator address, we will return error code 400 (`StatusBadRequest`) .